controllers/watcher_controller.go

```diff
@@ -392,6 +413,69 @@ func (r *WatcherReconciler) ensureMQ(
 	return transportURL, op, nil
 }
 
+func (r *WatcherReconciler) ensureKeystoneSvc(
+	ctx context.Context,
+	h *helper.Helper,
+	instance *watcherv1beta1.Watcher,
+	serviceLabels map[string]string,
+) (ctrl.Result, error) {
+	Log := r.GetLogger(ctx)
+	Log.Info(fmt.Sprintf("Reconciling the Keystone Service for '%s'", instance.Name))
+
+	//
+	// create Keystone service and user
+	//
+	hash, _, _, err := ensureSecret(
```

Comment:
I think I would move this call to the `Reconcile` method, so then we can set the `InputReadyCondition` to True and afterwards create the keystone service

Reply:
Good point, I will do it.